### ISSUE TITLE
Integrating the models in a React Native app

### ISSUE THREAD

**Issue by nuagoz** [ankandrew/fast-alpr#14]
Hello ankandrew,

I find your work remarkable, it gives me excellent results.
I would love to integrate it locally into a React Native application.

There is a library, [React Native Vision Camera](https://react-native-vision-camera.com/docs/guides/frame-processors), which supports frame processors for real-time processing.

There are also native Android and iOS plugins that enable various features:
[Frame Processor Plugins](https://react-native-vision-camera.com/docs/guides/frame-processor-plugins-community)

For example, [mrousavy/react-native-fast-tflite](https://github.com/mrousavy/react-native-fast-tflite) allows running any TensorFlow Lite model inside React Native.

I would really like to find a way to integrate your models and run them in React Native.
I tried converting .keras models to TensorFlow Lite (.tflite), but without success.

Do you have any insights or ideas on how to proceed? Does this seem feasible, or would it require a lot of adaptation?

Thank you for your help!

**Comment by snooplsm** [comment]
You will have to optimize the onnx models for mobile opset (ask chatgpt how to do that) if they are not already in that format.  Then you can use this library: 
 https://onnxruntime.ai/docs/get-started/with-javascript/react-native.html

Here is a demo of it working with Android Native:

https://github.com/user-attachments/assets/c84d3185-6f29-4cb1-9ece-5b53736aabfc


I am using opencv for modifying images though.

**Comment by nuagoz** [comment]
Thank you for your response and all the details @snooplsm. Your result is really impressive.

Yeah, I had seen that I could use the ONNX model with React Native, which is great. However, there is also a very efficient plugin if you provide TensorFlow Lite models: [https://github.com/mrousavy/react-native-fast-tflite](https://github.com/mrousavy/react-native-fast-tflite)

The problem is that I haven’t been able to successfully convert the models (ONNX or Keras) to TensorFlow Lite. The conversion process seems simple, but it just doesn’t work for me.

If you have any insights on the feasibility of this conversion or how to make it work, it would help me a lot.

Thanks again for your time!

**Comment by ankandrew** [comment]
Hi, @nuagoz! I've seen other people having problem with the conversion to `.tflite` from Keras/ONNX, although I haven't tried it myself. Could you try exporting (with random weights) MobileViT (**v1**) model and see if you can export it to `.tflite`? You can copy paste the model directly from Keras examples https://keras.io/examples/vision/mobilevit/. I can re-train that model with a bunch of data and officially support that model in fast-plate-ocr, for those who want to export it to TFLite.

**Comment by nuagoz** [comment]
Hi, @ankandrew. Thank you for your message. I actually tried exporting to `.tflite` using the `onnx2tf` tool with the following command:
```bash
onnx2tf -i yolo-v9-t-256-license-plates-end2end.onnx -o yolo-v9-t-256-license-plates-end2end.tflite
```

The conversion went through without errors, so I did get a `.tflite` model. However, I'm currently struggling to interpret the output correctly. I'm not sure how to process the results or whether the post-processing is the same as with the original model.

So honestly, if you were able to provide a working `.tflite` version, that would help me a lot. I'd definitely appreciate official support for that!

Let me know if I can help test anything.

**Comment by ankandrew** [comment]
Hi @nuagoz, the detector model your referring already includes NMS operation which alleviates the amount post-processing. To understand what each out represents checkout the source-code  - all libs are open-source :).  See:

https://github.com/ankandrew/open-image-models/blob/69007bba53eef71f19fecc083e412d8c4a7b6b88/open_image_models/detection/core/yolo_v9/postprocess.py#L6-L59

Now for the text recognition model, I'm planning to release soon another model that performs better (in terms of accuracy and speed), with more simple operations that the currently available one. Stay tuned for that!

Btw, I always like to visualize with https://netron.app/ the models to understand more input/outputs and the arch of the model, maybe that helps for further debugging the detector model.
